src/availability/query_data.rs

```diff
@@ -149,17 +163,19 @@ pub trait QueryablePayload: traits::BlockPayload {
     /// Get the transaction with a given hash, if it is in the block.
     fn transaction_by_hash(
         &self,
+        meta: &Self::Metadata,
```

Comment:
I'm curious, what exactly is this metadata?

Reply:
It's extra information produced when a block is built, which doesn't go in the payload itself but is required to interpret the payload: https://hotshot.docs.espressosys.com/hotshot/traits/trait.BlockPayload.html#tymethod.from_transactions.

For the sequencer, this is the namespace table, which ultimately goes in the block header.